pulp_rpm/app/models.py

```diff
@@ -447,7 +449,7 @@ class UpdateCollectionPackage(models.Model):
     epoch = models.TextField(blank=True)
     filename = models.TextField(blank=True)
     name = models.TextField(blank=True)
-    reboot_suggested = models.BooleanField(blank=True)
```

Comment:
Are you sure this is correct?  https://github.com/rpm-software-management/createrepo_c/blob/master/src/updateinfo.h#L48

Reply:
I think so. I think this is an optional piece of data so if it's not included createrepo_c returns None. This will be transformed to '' in the convert_createrepo_c_to_dict function with its `or ''` idiom. From there when saving Django raises an error saying that BooleanField must be True or False.

Could you try running this PR without this change?

Reply:
I think it would be better off just defaulting to false if not provided.

Reply:
Wait I think in the XML in some cases it just doesn't have a reboot suggested. If we default to False then we start introducing data. I think this is why createrepo_c returns None to start with. Is this right?

Reply:
Perhaps it could be a nullable boolean?  In which case, just ignore the ```or <x>``` bit entirely, just let it store a None.

In any case, we're wrapped around createrepo_c, and createrepo_c seems to be parsing the XML into a boolean regardless of whether it is or isn't present in the original.  At least, that's what it seems to do, I haven't dug deep into the code.  But I don't think it'll ever return None naturally.

The reason for the ```or ''``` is to avoid making a second "null" value for strings.  Then you have to deal with null strings *and* empty strings, which is needlessly painful.

Reply:
@bmbouter I think how it works is that the existence of the (self-closing) tag, itself, is the flag for true/false. 

https://github.com/rpm-software-management/createrepo_c/blob/72130e4d6500552c308cd7b06e5917c58770d03e/tests/testdata/updateinfo_files/updateinfo_01.xml#L24

Looking at the code, createrepo_c looks at the value of "reboot_suggested" and based on that, creates the XML node (or doesn't).  

Going by this, there's no way for a "null" value to be valid.  It's either true or false at all times.

Reply:
As a side note, I don't think any of our test fixtures have an errata where ```reboot_suggested``` is True.  At least not in my cursory lookthrough.

Reply:
I'm going to test this with a DB level default to False now.

Reply:
I switched it to `BooleanField(default=False)`. I also confirmed that at least with the fixuture data `getattr(package, CREATEREPO_UPDATE_COLLECTION_PACKAGE_ATTRS.REBOOT_SUGGESTED)` evaluates to `0`. I removed the `or ''` from the `createrepo_to_dict` method.

The smash tests pass locally so I'm going to push this now.